### PR TITLE
Fixes VSTS Bug 1009809: Save failed

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809Save
failed

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs [hunk 1]
```diff
@@ -426,6 +426,8 @@ async Task SaveAsyncInternal (SolutionItem entry)
 			try {
 				await entry.SaveAsync (monitor);
 				monitor.ReportSuccess (GettextCatalog.GetString ("Project saved."));
+			} catch (OperationCanceledException) {
+				monitor.Dispose ();
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs]: monitor.Dispose is redundant.

**Review by Therzok — CHANGES_REQUESTED** [review]

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs [hunk 2]
```diff
@@ -218,7 +218,12 @@ public static void SetParentToWindow (this Gtk.Window window, MonoDevelop.Compon
 		{
 			try {
 #if MAC
-				var nsWindow = parent.GetNativeWidget<NSWindow> ();
+				NSWindow nsWindow = null;
+				try {
+					nsWindow = parent.GetNativeWidget<NSWindow> ();
+				} catch {
+					; // nothing
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs]: We should at least log the exception.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs]: Not really it's kind of excepted and handled when nswindow == null.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs]: fixed.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs [hunk 3]
```diff
@@ -218,7 +218,12 @@ public static void SetParentToWindow (this Gtk.Window window, MonoDevelop.Compon
 		{
 			try {
 #if MAC
-				var nsWindow = parent.GetNativeWidget<NSWindow> ();
+				NSWindow nsWindow = null;
+				try {
+					nsWindow = parent.GetNativeWidget<NSWindow> ();
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs]: You need to check if the native widget is an via `parent.nativeWidget is NSWindow`. GetNativeWidget is expected to throw.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs]: That relies too much on the implementation details. I prefer doing it that way.

**Review by sevoku — APPROVED** [review]

**Review by Therzok — APPROVED** [review]

**Comment by sevoku** [comment]
@monojenkins rebase

**Review by bratsche — APPROVED** [review]

**Review by sevoku — APPROVED** [review]

**Pull request merged**